GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System.Collections;
+using System.Drawing;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace GenericsHomework
+{
+    public class Node<TNodeType> : ICollection<TNodeType>
+    {
+        private Node<TNodeType>? _Next { get; set; }
+
+        private TNodeType? _Value;
+
+        public TNodeType? Value
+        {
+            get {
+                return _Value; 
+            }
+            private set {
+                _Value = value;
+            }
+        }        
+        public Node<TNodeType> Next
+        {
+            get {
+                return _Next!; 
+            }
+            private set {
+                ArgumentNullException.ThrowIfNull(value);
+                _Next = value; 
+            }
+        }
+        public int Count
+        {
+            get {
+                int count = 0; 
+                foreach (TNodeType element in this) {
+                    count++; 
+                }
+                return count; 
+            }
+        }
+        bool ICollection<TNodeType>.IsReadOnly { get; } = false; 
+        public Node(TNodeType value)
+        {
+            Value = value;
+            Next = this; 
+        }
+
+        private Node(TNodeType value, Node<TNodeType> next) {
```

Comment:
```suggestion
        private Node(TNodeType value) {
```
Just set next to `this` when creating a new node.